packages/dds/tree/src/simple-tree/core/unhydratedFlexTree.ts

```diff
@@ -597,6 +597,11 @@ function getOrCreateField(
 		return new UnhydratedTreeSequenceField(parent.simpleContext, schema, key, parent, onEdit);
 	}
 
+	// TODO: this seems to used by unknown optional fields. They should probably use "optional" not "Forbidden" schema.
+	if (schema === FieldKinds.forbidden.identifier) {
```

Comment:
why is this all not just a switch statement?

Reply:
Really is should probably be a map lookup like I did with kindToClass in src/feature-libraries/flex-tree/lazyField.ts . I'm not sure why that pattern was not replicated when making the unhydrated variants of those classes. A switch would also be fine.